### ISSUE TITLE
Unable to run NVDA alpha

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#10833]
### Steps to reproduce:

Build and run NVDA alpha.

### Actual behavior:

Critical error. Here is the log:

<details> <summary>Log</summary>
INFO - __main__ (18:18:04.681) - MainThread (4004):
Starting NVDA version source-master-1e139d4
INFO - core.main (18:18:04.919) - MainThread (4004):
Config dir: D:\Cyrille\Dev\GIT\nvdaNew\source\userConfig
INFO - config.ConfigManager._loadConfig (18:18:04.919) - MainThread (4004):
Loading config: .\userConfig\nvda.ini
INFO - core.main (18:18:05.036) - MainThread (4004):
Using Windows version 10.0.18362 workstation
INFO - core.main (18:18:05.036) - MainThread (4004):
Using Python version 3.7.5 (tags/v3.7.5:5c02a39a0b, Oct 14 2019, 23:09:19) [MSC v.1916 32 bit (Intel)]
INFO - core.main (18:18:05.036) - MainThread (4004):
Using comtypes version 1.1.7
INFO - core.main (18:18:05.036) - MainThread (4004):
Using configobj version 5.1.0 with validate version 1.0.1
CRITICAL - __main__ (18:18:05.451) - MainThread (4004):
core failure
Traceback (most recent call last):
  File "source\nvda.pyw", line 215, in <module>
  File "core.py", line 256, in main
    import appModuleHandler
  File "appModuleHandler.py", line 27, in <module>
    import NVDAHelper
  File "NVDAHelper.py", line 19, in <module>
    import eventHandler
  File "eventHandler.py", line 9, in <module>
    import api
  File "api.py", line 19, in <module>
    import NVDAObjects.IAccessible
  File "NVDAObjects\IAccessible\__init__.py", line 22, in <module>
    import IAccessibleHandler
  File "IAccessibleHandler.py", line 18, in <module>
    import UIAHandler
  File "UIAHandler.py", line 11, in <module>
    from _UIAHandler import *  # noqa: F403
  File "_UIAHandler.py", line 29, in <module>
    from comtypes.gen import UIAutomationClient as UIA
ImportError: cannot import name 'UIAutomationClient' from 'comtypes.gen' (D:\Cyrille\Dev\GIT\nvdaNew\include\comtypes\comtypes\gen\__init__.py)

</details>

### Expected behavior:

NVDA running.

#### NVDA installed/portable/running from source:
Source

#### NVDA version:

Alpha (commit 1e139d4dad6e44cced4250365a6e75a1ac332c4b)

#### Windows version:

Win 10 1903

#### Name and version of other software in use when reproducing the issue:

Visual Studio 2019.

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Running other versions from source is also failing. I did not succeed in running from source since I have built with VS2019.

**Comment by josephsl** [comment]
Hi, did you recently do a scons –clean? If yes, you should do a git reset –hard before proceeding. Thanks.

**Comment by CyrilleB79** [comment]
Yes, I did a `scons -c` when trying to fix another issue.

source code is now starting up correctly. Thanks!

Just to understand, why do I need to do this `git reset --hard` after a `scons -c`?

**Comment by josephsl** [comment]
Hi, it’s because when scons –clean is run, it removes comInterfaces directory, which includes UIA modules. Thanks.

**Comment by CyrilleB79** [comment]
OK, but why does `scons --clean` clean some files that are fetched from the repo?

**Comment by Adriani90** [comment]
@CyrilleB79 are you still having this issue with last Alpha?

**Comment by CyrilleB79** [comment]
@Adriani90 no, as written in my previous comment, I can build and run correctly thanks to Joseph's comment.

However, this question remains unanswered:
> OK, but why does `scons --clean` clean some files that are fetched from the repo?

There are two possibilities:
* If it is a bug, it should be fixed
* If it is intended, it should be documented, maybe in the README of NVDA where you find build instructions.

@josephsl could you comment further? Thanks.

I will update the issue title and description.

**Comment by josephsl** [comment]
Hi, this is because SCons may not know about UIA COM interfaces files, so it removes them. I hope this can be resolved from SCons end by letting projects specify files to be kept when scons –clean is run. Thanks.

**Comment by CyrilleB79** [comment]
I do not know anything of these UIA COM interfaces files and why it would be necessary to tell expliciely to scons to keep them (on contrary to other source files).

Anyway, what do you suggest for this issue? May we open an issue against scons about this. Or we may link it here if there is already an issue opened against scons. Also may we add an information in the readme in the mean time?

**Comment by josephsl** [comment]
Hi, I think filing a feature suggestion with SCons would be the best option, as it will benefit NVDA and other projects. Thanks.

**Comment by CyrilleB79** [comment]
Would you mind to fill this feature suggestion agains scons project? I am not able to describe them the issue.

Indeed, again, I do not know the specificity of the UIA COM interfaces files and why they are deleted by `scons --clean`when I would have expected all repo source file to be kept without any specific configuration for them.
